src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -358,6 +358,19 @@ private function getAutowiredReference(TypedReference $reference): ?TypedReferen
             return $reference;
         }
 
+        if ($filterType && preg_match('/([&|])/', $type, $m)) {
```

Comment:
let's use `strpbrk` instead of `preg_match`:
`false !== $m = strpbrk($type, '&|')`